packages/suite/src/services/coingecko.ts

```diff
@@ -1,12 +1,56 @@
+/* eslint-disable max-classes-per-file */
 import { LastWeekRates, TickerId } from '@wallet-types/fiatRates';
 import FIAT_CONFIG from '@suite-config/fiat';
+import { differenceInMilliseconds } from 'date-fns';
 
+// const COINGECKO_API_BASE_URL = 'https://api.coingecko.com/api/v3';
 const COINGECKO_API_BASE_URL = 'https://cdn.trezor.io/dynamic/coingecko/api/v3';
 
 interface HistoricalResponse extends LastWeekRates {
     symbol: string;
 }
 
+class RateLimiter {
+    // Poor man's coingecko rate limiter
+    // Slows down requests so there is `delayMs` gap between them
+    public delayMs: number; // gap between each request
+    public lastFetchTimestamp: number;
+    private queued: number; // how many requests are waiting to be fired
+    private totalDelay: number; // by how much time are we gonna slown down next request
+
+    constructor(delayMs: number) {
+        this.delayMs = delayMs;
+
+        this.lastFetchTimestamp = 0;
+        this.queued = 0;
+        this.totalDelay = 0;
+    }
+
+    async limit(fn: any) {
```

Comment:
Any reason we escape the type-checking here with `any`? This function could be fully typed with generics
```suggestion
    async limit<T>(fn: () => Promise<T>): Promise<T> {
```

Reply:
thx